### ISSUE TITLE
Remove _AssertRaisesContext and assertRaises definitions from twisted/test/test_twisted.py

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5976]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#5976|
|Type|enhancement|
|Created|2012-09-13 19:11:52Z|
|Branch|https://github.com/twisted/twisted/tree/py26-assertraises-5976|

These are copied from the Python 2.7 standard library and are only necessary to support Python 2.6.

<details><summary>Searchable metadata</summary>

```
trac-id__5976 5976
type__enhancement enhancement
reporter__exarkun exarkun
priority__normal normal
milestone__EndOfLifePython2_6 EndOfLifePython2.6
branch__branches_py26_assertraises_5976 branches/py26-assertraises-5976
branch_author__hawkowl hawkowl
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1347563512000000 1347563512000000
changetime__1447414918511177 1447414918511177
version__None None
owner__hawkowl hawkowl

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_1"></a>|@hawkowl commented|
|-|-|

(In [45826]) Branching to py26-assertraises-5976.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_2"></a>|@hawkowl commented|
|-|-|

This is looking good. Please review.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_3"></a>|@adiroiban set owner to @hawkowl|
|-|-|

Look good. Thanks.

Please merge!

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_4"></a>|@hawkowl set status to `closed`|
|-|-|

(In [46183]) Merge py26-assertraises-5976: Remove assertRaises backport from Trial

Author: hawkowl
Reviewer: adiroiban
Fixes: [#5976](https://github.com/twisted/twisted/issues/5976)
